src/core/net/srp_server.cpp

```diff
@@ -438,15 +438,32 @@ void Server::CommitSrpUpdate(Error                    aError,
 
 void Server::Start(void)
 {
-    Error error = kErrorNone;
+    Error    error = kErrorNone;
+    uint16_t port  = kUdpPortMin;
 
     VerifyOrExit(!IsRunning());
 
+#if OPENTHREAD_CONFIG_SRP_SERVER_SAVE_INFO
+    Settings::SrpServerInfo info;
+    if (Get<Settings>().Read(info) == kErrorNone)
+    {
+        port = info.GetPort() + 1;
+        if (port < kUdpPortMin || port > kUdpPortMax)
+        {
+            port = kUdpPortMin;
+        }
+    }
```

Comment:
```suggestion
{
    Settings::SrpServerInfo info;
    
    if (Get<Settings>().Read(info) == kErrorNone)
    {
        port = info.GetPort() + 1;
        if (port < kUdpPortMin || port > kUdpPortMax)
        {
            port = kUdpPortMin;
        }
    }
}
```
Variable must be at the front, so...